src/components/Popover.vue

```diff
@@ -100,26 +107,44 @@ export default class Popover extends Vue {
     }
   }
 
+  handleEvent(event: any) {
+    switch (event.type) {
+      case 'click':
+        this.clickListener(event);
+        break;
+      case 'orientationchange':
+        this.updatePosition();
+        break;
+      case 'resize':
+        this.updatePosition();
+        break;
+      case 'scroll':
+        this.updatePosition();
+        break;
+    }
+  }
+
   setupPositioning() {
     // Move the popover into the body to prevent its hiding by an
     // `overflow: hidden` container
     document.body.appendChild(this.$el);
+    this.updatePosition();
     // Attach listeners
-    window.addEventListener('click', e => this.clickListener(e), { capture: true });
-    window.addEventListener('orientationchange', () => this.updatePosition());
-    window.addEventListener('resize', () => this.updatePosition());
+    window.addEventListener('click', this);
+    window.addEventListener('orientationchange', this);
+    window.addEventListener('resize', this);
     for (const parent of this.parents) {
-      parent.addEventListener('scroll', () => this.updatePosition());
+      parent.addEventListener('scroll', this);
```

Comment:
Passing the whole `this` is clever, as the component has indeed a method implementing `handleEvent`. Therefore, it does respect the [EventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventListener) interface.

However, this is a highly uncommon pattern, and it binds the entire vm reference to a DOMelement (idk if it can cause any problem of garbage collecting/reference counting).

I'm not against merging this, but I would find much simpler to simply bind the callbacks to `this` and passing them explicitely. that would look like:

```js
setupPositioning() {
  // These can also be defined only once in created or mounted hooks
  this.clickListener = () => {...}
  this.updatePosition = () => {...}

  window.addEventListener('click', this.clickListener);
  window.addEventListener('orientationchange', this.updatePosition);
  ...
}


destroyPositionning() {
  window.removeEventListener('click', this.clickListener);
  ...
}
```

Reply:
Done: https://github.com/ToucanToco/weaverbird/pull/534/files#diff-46e1611babaddbb455a4f41bfb470d75R123